webpack/assets/javascripts/react_app/components/Editor/EditorActions.js

```diff
@@ -0,0 +1,127 @@
+import API from '../../API';
+import { translate } from '../../common/I18n';
+
+import {
+  EDITOR_INITIALIZE,
+  EDITOR_CHANGE_STATE,
+  EDITOR_IMPORT_FILE,
+  EDITOR_REVERT_CHANGES,
+  EDITOR_EXEC_PREVIEW,
+  EDITOR_MODAL_TOGGLE,
+} from './EditorConstants';
+
+export const initializeEditor = props => dispatch => {
+  const {
+    data: { template, locked, type, hosts },
+    readOnly,
+    isMasked,
+    selectedView,
+    isRendering,
+    renderedValue,
+    showError,
+  } = props;
+
+  const initialState = {};
+
+  // initialize after changing editors
+  initialState.value = template || '';
+  if (readOnly !== locked) {
+    if (locked === true) initialState.readOnly = true;
+    else initialState.readOnly = false;
+  }
+  if (isMasked && type === 'templates') initialState.isMasked = false;
+  if (selectedView !== 'input') initialState.selectedView = 'input';
+  if (isRendering) initialState.isRendering = false;
+  if (renderedValue !== '') initialState.renderedValue = '';
+  if (showError) initialState.showError = false;
+
+  if (hosts && hosts.length !== 0) {
+    initialState.hosts = Object.assign(
+      {},
+      ...hosts.map(element => ({ [element.host.name]: element.host.id }))
+    );
+  }
+
+  dispatch({
+    type: EDITOR_INITIALIZE,
+    payload: { initialState },
```

Comment:
Can simply be: `payload: initialState`

Reply:
It's a pattern in all your actions, it's kind of nitpicking, but I see it as redundant.